internal/services/containers/kubernetes_cluster_node_pool_resource_test.go

```diff
@@ -955,6 +955,21 @@ func TestAccKubernetesClusterNodePool_nodeIPTags(t *testing.T) {
 	})
 }
 
+func TestAccKubernetesClusterNodePool_sourceSnapshotId(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_kubernetes_cluster_node_pool", "test")
+	r := KubernetesClusterNodePoolResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.sourceSnapshotId(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
```

Comment:
We can create the snapshot in the test using `data.CheckWithClientForResource`

```suggestion
	data.ResourceTest(t, r, []acceptance.TestStep{
		{
			Config: r.sourceSnapshotId(data),
			Check: acceptance.ComposeTestCheckFunc(
				check.That(data.ResourceName).ExistsInAzure(r),
				data.CheckWithClientForResource(func(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) error {
				
				// get the snapshot client here to create the snapshot
				
				}, data.ResourceName)
			),
		},
		data.ImportStep(),
	})
```